internal/gen/messages_set.go

```diff
@@ -222,6 +236,16 @@ nextField:
 func (g *generator) genMessageSetterFromFlags(message *protogen.Message) {
 	g.P("// SetFromFlags sets the ", message.GoIdent, " message from flags.")
 	g.P("func (m *", message.GoIdent, ") SetFromFlags(flags *", pflagPackage.Ident("FlagSet"), ", prefix string) (paths []string, err error) {")
+
+	// Message has semantical value just by existing
+	if len(message.Fields) == 0 {
+		g.P("if _, changed, err := ", flagspluginPackage.Ident("GetBool"), "(flags, ", flagspluginPackage.Ident("Prefix"), `("`, string(message.Desc.Name()), `", prefix)); err != nil {`)
+		g.P("return nil, err")
+		g.P("} else if changed", "{")
+		g.P("paths = append(paths, ", flagspluginPackage.Ident("Prefix"), `("`, string(message.Desc.Name()), `", prefix))`)
```

Comment:
It should not only add the path to the field mask, but also set the message.

Example:

```
message Trivial {}

message NonTrivial {
   Trivial trivial = 1 [
    (thethings.flags.field) = { select: true, set: true, semantical: true }
  ];
}
```

I expect that if I provide `--trivial`, and the base message is `NonTrivial`, I get a Go object that looks like:

```go
&NonTrivial{
  Trivial: &Trivial{}
}
```
And the field mask `[]string{"trivial"}`.

Does this make sense to you ?

Reply:
I get what you're going at. Basically have it return nil or `&object` based on the presence of the flag. I was at a bit of a loss yesterday as to which direction to take, hence the first iteration as too speak.

For the tags, what would the benefit be of having an extra `semantical` tag over the existing `set` tag?

Reply:
There are two reasons for that:
- I do not want to pollute the `set` parameter list with messages which are not semantical (i.e. having them as an empty message does not mean anything over having a null).
- I want to be able to add `semantical` to messages which are not empty. For example [`DynamicMode`](https://github.com/TheThingsNetwork/lorawan-stack/blob/389df361f8297ff392660a14b93e382be4d9f72f/api/end_device.proto#L204-L232) is made entirely of nullable fields (everything is a wrapper there, there are no integers/bools), and simply being in an empty `DynamicMode` has its own meaning (in this case, it means to use default values for every single setting of that message).

Reply:
Okay that makes sense. I've changed it to work based on the `semantical` flag.